cli/src/complete.rs

```diff
@@ -394,6 +394,44 @@ mod parse {
         }
         parse_flag(&["-f", "--from"], std::env::args())
     }
+
+    // Special parse function only for `jj log`. It has a --revisions flag,
+    // instead of the usual --revision, and it can be supplied multiple times.
+    // The default revset for log _with specified paths_ is 'all()', so it
+    // would be most "correct" to use that as the default. However, that is
+    // terrible for performance. Instead, we just complete the files in "@".
+    // If the user still wants to have completions for every file that has
+    // ever existed in the repository, they can still provide -r=all().
+    pub fn log_revision() -> String {
+        let candidates = &["-r", "--revisions"];
+        let mut args = std::env::args();
+        let mut revisions = Vec::new();
+
+        while let Some(arg) = args.next() {
+            if candidates.contains(&arg.as_ref()) {
+                match args.next() {
+                    Some(rev) => revisions.push(rev),
+                    None => continue,
+                }
+            }
+            let Some((flag, rev)) = arg.split_once('=') else {
+                continue;
+            };
```

Comment:
Share some code with `parse_flag()` here? Perhaps a function that takes candidates and a mutable reference to an iterator would work?